src/backup.ts

```diff
@@ -0,0 +1,387 @@
+ */
+type IBackupTranslatedEnum = TranslateEnumKeys<
+  databaseAdmin.spanner.admin.database.v1.IBackup,
+  'state',
+  typeof databaseAdmin.spanner.admin.database.v1.Backup.State
+>;
+
+export type GetMetadataResponse = [IBackupTranslatedEnum];
+type GetMetadataCallback = RequestCallback<IBackupTranslatedEnum>;
+
+type UpdateExpireTimeCallback = RequestCallback<Backup>;
+
+type DeleteCallback = RequestCallback<void>;
+
+/**
+ * The {@link Backup} class represents a Cloud Spanner backup.
+ *
+ * Create a `Backup` object to interact with or create a Cloud Spanner backup.
+ *
+ * @class
+ *
+ * @example
+ * const {Spanner} = require('@google-cloud/spanner');
+ * const spanner = new Spanner();
+ * const instance = spanner.instance('my-instance');
+ * const backup = instance.backup('my-backup');
+ */
+class Backup {
+  request: <T, R = void>(
+    config: RequestConfig,
+    callback: RequestCallback<T, R>
+  ) => void;
+
+  formattedName_: string;
+
+  constructor(
+    private instance: Instance,
+    private backupId: string,
+    private databasePath?: string,
+    private expireTime?: PreciseDate
+  ) {
+    this.request = instance.request;
+    this.formattedName_ =
+      this.instance.formattedName_ + '/backups/' + this.backupId;
+  }
+
+  create(): Promise<CreateBackupResponse>;
+  create(callback: CreateBackupCallback): void;
+  /**
+   * Create a backup.
+   *
+   * @method Backup#create
+   * @returns {Promise<CreateBackupResponse>} when resolved, the backup
+   *     operation will have started, but will not have necessarily completed.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const database = spanner.database('my-database');
+   * const oneDay = 1000 * 60 * 60 * 24;
+   * const ExpiryTime = new PreciseDate(Date.now() + oneDay)
+   * const backup = instance.backup('my-backup', database.formattedName_, expiryTime);
+   * const [, backupOperation] = await backup.create();
+   * // Await completion of the backup operation.
+   * await backupOperation.promise();
+   */
+  create(
+    callback?: CreateBackupCallback
+  ): Promise<CreateBackupResponse> | void {
+    if (!this.expireTime) {
+      throw new Error('Expire time is required to create a backup.');
+    }
+    if (!this.databasePath) {
+      throw new Error('Database path is required to create a backup.');
+    }
+
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.ICreateBackupRequest = extend(
+      {
+        parent: this.instance.formattedName_,
+        backupId: this.backupId,
+        backup: {
+          database: this.databasePath,
+          expireTime: this.expireTime.toStruct(),
+          name: this.formattedName_,
+        },
+      }
+    );
+    return this.request<CreateBackupGaxOperation>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'createBackup',
+        reqOpts,
```

Comment:
In addition to `reqOpts`, I believe we can support "gaxOptions". This would allow finer control over gax's internal configurations. An example from Database: https://github.com/googleapis/nodejs-spanner/blob/c1098c5f4509918cacd3942b8f09354c88a85bb9/src/database.ts#L1124

Reply:
Note that this applies wherever we make an API request. Where `reqOpts` is accepted, so is `gaxOptions`.

Reply:
All done. Thanks for noticing this.

Reply:
Thank you! I think it needs to be added to the documentation for those methods as well.

Reply:
Sorry about that. Fixed now.